### PR TITLE
Clarified North/South, removed anchors, added code highlights

### PR DESCRIPTION
**Pull request by jason-fox**
After discussion with @dcalvoalonso 

With standard English usage, the “bound”  suffix in implies a direction e.g “Homeward bound" means "going **towards** my home”, therefore “Southbound” means "going **towards** the South”
In contrast, the phrase "to the North of” does not imply direction, it implies relative location — you can be to the North of a town, but travelling Southbound - i.e. towards the city.
The main common usage of Northbound/Southbound is found with motorway traffic reports e.g. “the Northbound carriageway is blocked"

Therefore the following usage is correct common English, and the existing documents should be corrected to use it: 

* Traffic **to the South of** the IoT Agent refers to the exchange of information and commands between a device and an IoT Agent (Native protocols)
Traffic **to the North of** the IoT Agent traffic considers the link between the IoT Agent  and the CB.  (NGSI)
* **Southbound** traffic refers to the commands sent from the CB down to the device
* **Northbound** traffic refers to the measurements sent up from the IoT device to the CB
* **Northbound** traffic also refers to the results of commands  sent up from the IoT device to the CB

I have updated the documents  and removed `<a>`  tags and added code highlights for ReadTheDocs whilst updating.

### CODE REVIEW

File: docs/stepbystep.md [hunk 1]
```diff
@@ -136,74 +143,81 @@ check the documentation to see how to connect the IoTA to a MongoDB instance.
...
 mosquitto_pub -h <mosquittoIp> -t /<apiKey>/<devId>/attrs -m '{"L":4,"T": "31.5","H":30}'
 ```
 
 If you execute both commands in different windows, when you run the latter command, you should see the string `{"L":4,"T": "31.5","H":30}`
 appearing in the former.
 
-This tutorial will use mainly the mosquitto clients and curl commands, to give a detailed view of how the APIs (both 
-northbound and southbound work). Whenever a command is needed, the exact command to be executed will be given.
+This tutorial will use mainly the mosquitto clients and curl commands, to give a detailed view of how the APIs (
+NGSI traffic north of the IoT Agent and JSON south of the IoT Agent). Whenever a command is needed, the exact command to be executed will be given.
```
Comment (dcalvoalonso) [inline docs/stepbystep.md]: I would rather say _NGSI traffic north of the IoT Agent and MQTT south of the IoT Agent._ JSON is not a communication protocol but a data model.

File: docs/operations.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ while the 'Severity' criterium is as follows:
 * **Major** - The system has a problem that degrades the service and must be addressed
 * **Warning** - It is happening something that must be notified
 
-## <a name="errorcode"/> Error naming code
+## Error naming code
```
Comment (dcalvoalonso) [inline docs/operations.md]: Now the index link to this subsection does not work

File: docs/stepbystep.md [hunk 3]
```diff
@@ -64,48 +64,55 @@ work as well (previous versions could also work, but also may not, so we encoura
 To enhance readability all the commands will be executed as root. To use other users, give it the appropriate permissions
 and use `sudo` as usual (installation from the RPM package creates a special user for the agent, but it will not be
 used along this tutorial).
- 
-## <a name="singledevice">Provisioning a single device with the default API Key</a>
+
+## Provisioning a single device with the default API Key
```
Comment (dcalvoalonso) [inline docs/stepbystep.md]: Now the index link to this subsection does not work

File: docs/stepbystep.md [hunk 4]
```diff
@@ -397,15 +418,15 @@ We will get something like this:
 
 That shows the information we sent with the measures has been written to the Context Broker properly.
 
-## <a name="acls">Using ACLs to secure provisioning access</a>
+## Using ACLs to secure provisioning access
```
Comment (dcalvoalonso) [inline docs/stepbystep.md]: Now the index link to this subsection does not work

File: docs/usermanual.md [hunk 5]
```diff
@@ -9,7 +9,7 @@
 * [Development documentation](#development)
 * [New transport development](#newtransport)
 
-## <a name="apioverview"/> API Overview
+## API Overview
```
Comment (dcalvoalonso) [inline docs/usermanual.md]: Now the index link to this subsection does not work

File: docs/usermanual.md [hunk 6]
```diff
@@ -23,15 +23,16 @@ by the device to the agent. Passive or lazy attributes, i.e. those attributes wh
 request from the agent, are not implemented. Please check the issue
 [#89](https://github.com/telefonicaid/iotagent-json/issues/89) for more details and updates regarding its implementation.
 
-### <a name="httpbinding"/> HTTP binding
+###  HTTP binding
```
Comment (dcalvoalonso) [inline docs/usermanual.md]: Now the index link to this subsection does not work

File: docs/usermanual.md [hunk 7]
```diff
@@ -313,7 +315,7 @@ in a transformation expression in the IoTAgent). In this cases, when a reverse e
 bidirectional attribute is modified, the IoTAgent sends a command to the original device, with the name defined in the
 reverse expression attribute and the ID of the device (see Commands Syntax, just above).
 
-#### <a name="amqpbinding"/> AMQP binding
+#### AMQP binding
```
Comment (dcalvoalonso) [inline docs/usermanual.md]: Now the index link to this subsection does not work

File: docs/usermanual.md [hunk 8]
```diff
@@ -361,7 +363,7 @@ fields codified in hexadecimal with a fixed 4 character length, without comma se
 of this attribute, adding the "batteryType" and "percentage" fields to the entity.
 
 
-##  <a name="development"/> Development documentation
+##  Development documentation
```
Comment (dcalvoalonso) [inline docs/usermanual.md]: Now the index link to this subsection does not work

File: docs/usermanual.md [hunk 9]
```diff
@@ -486,7 +490,7 @@ This command will only work after the developer has executed init-dev-env (that'
 
 This command will also launch the coverage, doc and complexity task (see in the above sections).
 
-## <a name="newtransport"/> New transport development
+## New transport development
```
Comment (dcalvoalonso) [inline docs/usermanual.md]: Now the index link to this subsection does not work

Comment (jason-fox) [reply docs/usermanual.md]: Fixed - facd80d - ToC regenerated

File: docs/usermanual.md (hunk 8 above)
Comment (jason-fox) [reply docs/usermanual.md]: Fixed - facd80d - ToC regenerated

File: docs/usermanual.md (hunk 7 above)
Comment (jason-fox) [reply docs/usermanual.md]: Fixed - facd80d - ToC regenerated

File: docs/usermanual.md (hunk 6 above)
Comment (jason-fox) [reply docs/usermanual.md]: Fixed - facd80d - ToC regenerated

File: docs/usermanual.md (hunk 5 above)
Comment (jason-fox) [reply docs/usermanual.md]: Fixed - facd80d - ToC regenerated

File: docs/stepbystep.md (hunk 4 above)
Comment (jason-fox) [reply docs/stepbystep.md]: Fixed - facd80d - ToC regenerated

File: docs/stepbystep.md (hunk 3 above)
Comment (jason-fox) [reply docs/stepbystep.md]: Fixed - facd80d - ToC regenerated

File: docs/operations.md (hunk 2 above)
Comment (jason-fox) [reply docs/operations.md]: Fixed - facd80d - ToC regenerated

File: docs/stepbystep.md (hunk 1 above)
Comment (jason-fox) [reply docs/stepbystep.md]: Why not both? Fixed  5de424b - *... the APIs - NGSI traffic north of the IoT Agent and MQTT (with a JSON payload) south of the IoT Agent.*

**Pull request merged**